public/app/features/explore/Explore.tsx

```diff
@@ -182,6 +182,30 @@ export class Explore extends React.PureComponent<Props, ExploreState> {
     }
   };
 
+  inspectQuery = async (check: string, key: string, value: string) => {
+    let isActive = false;
+    const { queries } = this.props;
+    for (const query of queries) {
+      const ds = await getDataSourceSrv().get(query.datasource);
+      if (ds.inspectQuery) {
+        isActive = ds.inspectQuery(query, {
+          check,
+          attributes: {
+            key,
+            value,
+          },
+        });
+      } else {
+        return false;
```

Comment:
Not sure if that's gonna work with mixed data sources, e.g. if a Prometheus query was first, followed by a Loki query it'd just always return false 🤔

Reply:
For the moment that would be acceptable.

Reply:
To further ellaborate, single datasource covers most of our use cases for this feature.

Reply:
It is, because as `modifyQuery` originally worked (and still works), it adds the filter (positive or negative) to all the queries, not just the one that produced the log line.